kafka-ballerina/tests/consumer_client_tests.bal

```diff
@@ -0,0 +1,1088 @@
+// Copyright (c) 2020 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/lang.'string;
+import ballerina/log;
+import ballerina/lang.runtime as runtime;
+import ballerina/test;
+
+const DOCKER_COMPOSE_FILE = "docker-compose.yaml";
+const TEST_MESSAGE = "Hello, Ballerina";
+const EMPTY_MEESAGE = "";
+const TEST_DIRECTORY = "";
+const decimal TIMEOUT_DURATION = 5;
+const decimal DEFAULT_TIMEOUT = 10;
+
+string topic1 = "test-topic-1";
+string topic2 = "test-topic-2";
+string topic3 = "test-topic-3";
+string topic4 = "test-topic-4";
+string topic5 = "test-topic-5";
+string emptyTopic = "empty-topic";
+string nonExistingTopic = "non-existing-topic";
+string manualCommitTopic = "manual-commit-test-topic";
+
+string receivedMessage = "";
+string receivedMessageWithCommit = "";
+string receivedMessageWithCommitOffset = "";
+string receivedConfigMessage = "";
+//string saslMsg = "";
+
+int receivedMsgCount = 0;
+
+TopicPartition nonExistingPartition = {
+    topic: nonExistingTopic,
+    partition: 999
+};
+
+ProducerConfiguration producerConfiguration = {
+    clientId: "basic-producer",
+    acks: ACKS_ALL,
+    maxBlock: 6,
+    requestTimeout: 2,
+    retryCount: 3
+};
+Producer producer = check new (DEFAULT_URL, producerConfiguration);
+
+@test:Config {}
+function consumerServiceTest() returns error? {
+    check sendMessage(TEST_MESSAGE.toBytes(), topic1);
+    ConsumerConfiguration consumerConfiguration = {
+        topics: [topic1],
+        offsetReset: OFFSET_RESET_EARLIEST,
+        groupId: "consumer-service-test-group",
+        clientId: "test-consumer-1"
+    };
+    Listener consumer = check new (DEFAULT_URL, consumerConfiguration);
+    check consumer.attach(consumerService);
+    check consumer.'start();
+
+    runtime:sleep(7);
+    test:assertEquals(receivedMessage, TEST_MESSAGE);
+}
+
+@test:Config {}
+function consumerServiceSubscribeErrorTest() returns error? {
+    check sendMessage(TEST_MESSAGE.toBytes(), topic1);
+    ConsumerConfiguration consumerConfiguration = {
+        topics: [topic1],
+        offsetReset: OFFSET_RESET_EARLIEST,
+        clientId: "test-consumer-2"
+    };
+    Listener?|error? result = trap new (DEFAULT_URL, consumerConfiguration);
```

Comment:
`Listener?|error?` if you read this, (listener or nil or error or nil) you see this has nil twice :) 
Also no need to use `trap` here. This can simply be written as,
`Listener|Error result = new (DEFAULT_URL, consumerConfiguration);`